### ISSUE TITLE
Generate unique IDs for form elements

### ISSUE THREAD

**Issue by susickypavel** [fabian-hiller/modular-forms#99]
As mentioned in #92.

It might be helpful to generate a unique id per form that can be used for pairing form + its form elements (inputs, selects, etc.).

When I wrote about it in #92, I thought that the `createForm` function could expose it, and users could use it or use some of their own ID if they wish, but I also think that if the id would come with the props from the `Field` component, it could also be very useful.

```tsx
<Field name="email" validate={…}>
  {(field, props) => (
    <div>
      <label htmlFor={field.id}>Email</label> // Linking label with ID
      <input
        id={field.id} // Field ID
        aria-describedby={`${field.id}-error`} // Error element ID
        value={field.value.value || ''}
        type="email"
        required
        {...props}
      />
      {field.error && <div id={`${field.id}-error`}>{field.error}</div>} // Field ID+ suffix (-error)
    </div>
  )}
</Field>
```

In SolidJS & React it shouldn't be even hard to implement since they have stable `createUniqueID` and `useId` functions between server/browser and hydrating.

I'm not familiar with Qwik in this regard.

**Comment by fabian-hiller** [comment]
Good idea! Currently I think it makes sense to add it to the `props` object.
